### PR TITLE
Examples+main improvements

### PR DESCRIPTION
**Pull request by dccote**
`python -m raytracing` has always bugged me: it is hard to maintain and the code is hidden in `__main__`. Adding new code requires quite a bit of work.

I took the code out of `__main__` and created a submodule `examples`.  The code can actually be imported easily with:

```
import examples

examples.ex01.exemplecode()
```

examples will load all the files starting with `ex` in the directory. If people use `python -m raytracing`, then the actual source code will be displayed on the same figure in place of comments, which I think looks better than before:

image

It also prints (with syntax coloring) the code to the console.
image

You can get a list of all available examples with a description with `python -m raytracing -l`:

```
All example code on your machine is found at: /Users/dccote/GitHub/RayTracing/raytracing/examples
 1. ex01.py A single lens f = 50 mm, infinite diameter
 2. ex02.py Two lenses, infinite diameters
 3. ex03.py Finite-diameter lens
 4. ex04.py Aperture behind lens
 5. ex05.py Simple microscope system
[...]
```
@PyMarc2 @JLBegin What do you guys think? You may need to install pygments.

### CODE REVIEW

**Comment by dccote** [comment]
I am waiting for you to release 1.3.5 :-)

File: examples/twoLenses.py
Comment (PyMarc2) [inline examples/twoLenses.py]: pep8: end of file requires an empty line

File: raytracing/__main__.py
Comment (PyMarc2) [inline raytracing/__main__.py:45]: Why `elif` ? there doesn't seem to be a constraint that prohibits the user of doing `-r -c`. If that is the case, we should see a `MutuallyExclusiveGroup` as the argparse argument parser class

File: raytracing/__main__.py
Comment (PyMarc2) [inline raytracing/__main__.py]: would it be `if examples:`
and removing all the other examples in the __main__ ?

File: raytracing/__main__.py
Comment (PyMarc2) [inline raytracing/__main__.py:67]: perfect, last comment was on outdated commit. That is approved.

File: raytracing/examples/__init__.py
Comment (PyMarc2) [inline raytracing/examples/__init__.py:30]: clearer as f strings ? `with open(f"{topDir}/{file}") as f:`

File: raytracing/examples/__init__.py
Comment (PyMarc2) [inline raytracing/examples/__init__.py:34]: `srcCode = ''.join(srcCode)` What for ?

**Review by PyMarc2 — APPROVED** [review]
I approve of these changes, even though the _init_ might be hard to read for some, it is worth it for the clarity of the _main_. 
However, I still prefer the way it looked with the HTML style with my old branch. The main is better managed, but the look and feel are worse, in my opinion. The title was coloured and pretty and everything seemed more "together".

I inserted a few comments that would need attention before merging

File: raytracing/__main__.py
Comment (dccote) [reply raytracing/__main__.py:45]: It makes more sense to have only one. Not sure how to make a group.

File: raytracing/__main__.py
Comment (dccote) [reply raytracing/__main__.py]: Yes I think you had an old version

File: raytracing/__main__.py
Comment (dccote) [reply raytracing/__main__.py]: I just modified it and it gets weird because -e defaults to all so we don't see the list, we just see a window popping up.  I am leaving the elif

**Pull request merged**
